### DISCUSSION TITLE
[Feature Request / Idea] Insert markdown from filepath or URL

### DISCUSSION THREAD

**Discussion by couleurm** [squidfunk/mkdocs-material#6478] [Q&A] [answered]
Let's theoretically say I want to include https://rentry.co/NVIDIAOptimize/raw in my guide, is there any javascript tricks/features to have it try to grab some markdown text and feed it to MkDocs?

Another solution would be a script that runs on GitHub actions and parses for a special syntax and replace it with whatever it requests, I already did some similar trick to do extra redirects [here](https://github.com/couleur-tweak-tips/CTT/blob/50cf8e2df6067ee21c4962106fb18aa6508b4830/redirector.ps1)

**Comment by kamilkrzyskow** [comment, marked as answer]
MkDocs and Markdown processing is made via Python, I don't think JavaScript will help you here.

You could use Snippets to download and include Markdown content in an already existing page.
https://facelessuser.github.io/pymdown-extensions/extensions/snippets/

However if you want to download Markdown and make any changes to it, some additional preprocessing I think you would have to write a hook that downloads it first and makes appropriate changes.
https://www.mkdocs.org/user-guide/configuration/#hooks
